File: mobile_app/lib/models/review.dart
/// A customer review on a product.
class Review {
  Review({
    required this.id,
    required this.rating,
    this.title,
    this.body,
    this.customerName,
    this.isVerifiedPurchase = false,
    this.date,
    this.photos = const [],
  });

  final int id;
  final int rating;
  final String? title;
  final String? body;
  final String? customerName;
  final bool isVerifiedPurchase;
  final DateTime? date;
  final List<String> photos;

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'] as int,
      rating: (json['rating'] as num?)?.toInt() ?? 5,
      title: json['title'] as String?,
      body: json['body'] as String?,
      customerName: json['customer_name'] as String?,
      isVerifiedPurchase: json['is_verified_purchase'] as bool? ?? false,
      date: (json['review_date'] ?? json['created_at']) != null
          ? DateTime.tryParse((json['review_date'] ?? json['created_at']) as String)
          : null,
      photos: ((json['photos'] as List<dynamic>?) ?? const [])
          .map((e) => e as String)
          .toList(),
    );
  }
}

File: backend/app/Models/Review.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Spatie\MediaLibrary\HasMedia;
use Spatie\MediaLibrary\InteractsWithMedia;
use Spatie\MediaLibrary\MediaCollections\Models\Media;

class Review extends Model implements HasMedia
{
    use HasFactory;
    use InteractsWithMedia;

    public function registerMediaCollections(): void
    {
        $this->addMediaCollection('photos')
            ->useDisk('original_images')
            ->storeConversionsOnDisk('public');
    }

    public function registerMediaConversions(?Media $media = null): void
    {
        $this->addMediaConversion('thumb')->width(400)->format('png')->quality(78);
    }

    protected $fillable = [
        'product_id',
        'user_id',
        'order_id',
        'customer_name',
        'customer_email',
        'rating',
        'title',
        'body',
        'status',
        'is_verified_purchase',
        'review_date',
    ];

    protected function casts(): array
    {
        return [
            'rating' => 'integer',
            'is_verified_purchase' => 'boolean',
            'review_date' => 'date',
        ];
    }

    /**
     * The date to show on the storefront — an admin-set review_date (for
     * backdating an admin-authored review) takes priority over created_at.
     */
    public function displayDate(): \Illuminate\Support\Carbon
    {
        return $this->review_date ?? $this->created_at;
    }

    public function product(): BelongsTo
    {
        return $this->belongsTo(Product::class);
    }

    public function order(): BelongsTo
    {
        return $this->belongsTo(Order::class);
    }

    public function scopeApproved(Builder $query): Builder
    {
        return $query->where('status', 'approved');
    }
}
